dist/tools/doccheck/check.sh

```diff
@@ -74,3 +75,40 @@ then
     echo "${UNDEFINED_GROUPS_PRINT}"
     exit 2
 fi
+
+# Check for groups defined multiple times:
+# - remove empty spaces from the beginning of the line
+# - replace spaces with ':' for easily looping on the result when printing
+# - lines starting with '1:' are excluded
+MULTIPLE_DEFINED_GROUPS=$(echo "${DEFINED_GROUPS}" | uniq -c | \
+    sed 's/^ *//' | tr ' ' ':' | grep -v "^1:")
+
+# Remove the group description from each line containing a group definition.
+# Examples of input lines:
+# boards/nucleo-l152re/doc.txt:@defgroup boards_nucleo-l152re STM32 Nucleo-L152RE
+# boards/nucleo-l152re/include/periph_conf.h: * @defgroup boards_nucleo-l152re
+# Expected output lines patterns:
+# <filepath>: * <group name>
+# <filepath>:<group name>
+ALL_DEFGROUP_CLEANED=$(echo "${ALL_RAW_DEFGROUP}" | \
+    awk -F@ '{ split($2, end, " "); printf("%s%s\n",$1,end[2]) }')
+
+MULTIPLE_GROUPS_PRINT=$( \
+    for group in ${MULTIPLE_DEFINED_GROUPS}; \
+    do \
+        _group_name=$(echo ${group} | awk -F: '{ printf ("%s\n",$2) }'); \
+        _group_count=$(echo ${group} | awk -F: '{ printf ("%s\n",$1) }'); \
+        echo -e "\n${CWARN}${_group_name}${CRESET} defined ${CWARN}${_group_count}${CRESET} times in:"; \
+        echo "${ALL_DEFGROUP_CLEANED}" | grep "\<${_group_name}\>$" | sort -u | \
+            awk -F: '{ print "\t" $1 }'; \
+    done \
+    )
+
+if [ -n "${MULTIPLE_DEFINED_GROUPS}" ]
+then
+    COUNT=$(echo "${MULTIPLE_DEFINED_GROUPS}" | wc -l)
+    echo -ne "${CERROR}ERROR${CRESET} "
+    echo -e "There are ${CWARN}${COUNT}${CRESET} Doxygen groups defined multiple times:"
+    echo "${MULTIPLE_GROUPS_PRINT}"
```

Comment:
Why not just put the loop here instead of assigning its result to a variable you print anyway?